src/alire/alire-test_runner.adb

```diff
@@ -294,11 +299,13 @@ package body Alire.Test_Runner is
          --  Helper function to append all .adb files in a tree
          --  to the `Test_List` vector
 
-         Name : constant String :=
-           Strip_Prefix
-             (This.Path, Prefix => (Root.Path / "src") & OS_Lib.Dir_Separator);
+         Name : constant OS_Lib.Portable_Path_Like :=
+           OS_Lib.To_Portable
+             (Strip_Prefix
+                (This.Path,
+                 Prefix => (Root.Path / "src") & OS_Lib.Dir_Separator));
 
-         Filtering_Name : constant String :=
+         Filtering_Name : constant OS_Lib.Portable_Path_Like :=
            AAA.Strings.Replace
              (Text => Name, Match => Root_Prefix (Root), Subst => "");
```

Comment:
This breaks if we have a structure like

```
src
|_ crate_tests-for_fun
   |_ crate_tests-example.adb
```

Which admittedly is a bit contrived but still.

Reply:
In pseudo Ada:

```ada
function Display_Name (Name : Portable_Path; Root_Prefix : String)
return Portable_Path_Like is
   Simple : String := Strip_Prefix (Simple_Name (Name), Root_Prefix);
   Parent_Name : String := Portable_Path (Parent (Name));
begin
   if Parent_Name = "." then
      return Simple;
   else
      return Parent_Name & "/" & Simple;
   end if;
end Display_Name;
```

And you could reuse the function in the spawn code.

Reply:
https://github.com/AldanTanneo/alire/commit/4d682b19f3cd486ef97d06e8979723b3e976d7e5

My commit that fixes this (and some other structural things)

Reply:
Cool, thanks.

Reply:
> Which admittedly is a bit contrived but still.

Ugh, right.